ursa.ino

```diff
@@ -258,83 +333,73 @@ void zeroMPU6050() {//find how much offset each gyro axis has to zero out drift.
+float parseFl() {//return float from 4 bytes starting at wifI position in recvdData
+  union {//this lets us translate between two variable types (equal size, but one's 4 bytes in an array, and one's a 4 byte float) Reference for unions: https://www.mcgurrin.info/robots/127/
     byte b[4];
     float v;
   } d;//d is the union, d.b acceses the byte array, d.v acceses the float
-  d.b[0] = *(arrayPointer + *i);
-  i++;
-  d.b[1] = *(arrayPointer + *i);
-  i++;
-  d.b[2] = *(arrayPointer + *i);
-  i++;
-  d.b[3] = *(arrayPointer + *i);
-  i++;
+  d.b[0] = recvdData[wifI];
+  wifI++;
+  d.b[1] = recvdData[wifI];
+  wifI++;
+  d.b[2] = recvdData[wifI];
+  wifI++;
+  d.b[3] = recvdData[wifI];
+  wifI++;
   return d.v;
 }
-void arrayBl(boolean msg, byte * arrayPointer, int* i) { //declare a function that returns a boolean and will be given the location of an array and what element of the array to start at
-  if (msg) {
-    *(arrayPointer + *i) = 1;//set the ith element of the array to 1
-  } else {
-    *(arrayPointer + *i) = 0;
-  }
-  i++;
+void sendBl(boolean msg) {//add a boolean to the tosendData array
+  tosendData[wifI] = msg;
+  wifI++;
 }
-void arrayBy(byte msg, byte * arrayPointer, int* i) { //declare a function that returns a byte and will be given the location of an array and what element of the array to start at
-  *(arrayPointer + *i) = msg;
-  i++;
+void sendBy(byte msg) {//add a byte to the tosendData array
+  tosendData[wifI] = msg;
+  wifI++;
 }
-void arrayIn(int msg, byte * arrayPointer, int* i) { //declare a function that returns an int and will be given the location of an array and what element of the array to start at
+void sendIn(int msg) {//add an int to the tosendData array (two bytes)
   union {
     byte b[2];
     int v;
   } d;//d is the union, d.b acceses the byte array, d.v acceses the int
   d.v = msg;//put the value into the union as an int
-  *(arrayPointer + *i) = d.b[0];
-  i++;
-  *(arrayPointer + *i) = d.b[1];
-  i++;
+  tosendData[wifI] = d.b[0];
+  wifI++;
+  tosendData[wifI] = d.b[1];
+  wifI++;
 }
-void arrayFl(float msg, byte * arrayPointer, int* i) { //declare a function that returns a (4 byte) float and will be given the location of an array and what element of the array to start at
+void sendFl(float msg) {//add a float to the tosendData array (four bytes)
   union {//this lets us translate between two variables (equal size, but one's 4 bytes in an array, and one's a 4 byte float Reference for unions: https://www.mcgurrin.info/robots/127/
     byte b[4];
     float v;
   } d;//d is the union, d.b acceses the byte array, d.v acceses the float
   d.v = msg;
-  *(arrayPointer + *i) = d.b[0];
-  i++;
-  *(arrayPointer + *i) = d.b[1];
-  i++;
-  *(arrayPointer + *i) = d.b[2];
-  i++;
-  *(arrayPointer + *i) = d.b[3];
-  i++;
+  tosendData[wifI] = d.b[0];

```

Comment:
Is the correct order for a 4 byte variable? i'm not familiar with the receiving code.

Reply:
It is correct because the receiving code hasn't been written yet :smile: If @jamdan2 would like me to change the order, I would be happy to.

Reply:
💯